### PR TITLE
Fixes some documentation compilation errors.

### PR DESCRIPTION
**Pull request by jfinkels**
This should fix some of the errors in #1667.

One of the main issues here was that some section headings had one too many hyphens on the following line. For example:

    Notes
    ------

will cause an error. It must be

    Notes
    -----

### CODE REVIEW

File: networkx/algorithms/bipartite/projection.py [hunk 1]
```diff
@@ -334,17 +339,18 @@ def overlap_weighted_projected_graph(B, nodes, jaccard=True):
     --------
     >>> from networkx.algorithms import bipartite
     >>> B = nx.path_graph(5)
-    >>> G = bipartite.overlap_weighted_projected_graph(B, [0, 2, 4])
+    >>> projected = bipartite.overlap_weighted_projected_graph
+    >>> G = projected(B, [0, 2, 4])
     >>> list(G)
     [0, 2, 4]
-    >>> list(G.edges(data=True))
+    >>> G.edges(data=True)
```
Comment (SanketDG) [inline networkx/algorithms/bipartite/projection.py]: Should be `list(G.edges(data=True))` as per 2.0 api

**Comment by hagberg** [comment]
There are also a lot of other changes besides the underlines - adding automodule directives and switching from authofunction to autoclass.   Are those from errors or changing the format/function?

**Comment by jfinkels** [comment]
Adding `automodule` seems to be necessary, see for example the current documentation for `algorithms.shortest_path` versus the current documentation for `algorithms.centrality`: the former is generated correctly, the latter is not. Without it, it seems Sphinx can't figure out where the functions live.

I'm actually not sure if changing `autofunction` to `autoclass` would change anything, but it seemed appropriate, since those are actually classes, not functions. Was it like that for a reason? There were a lot of errors concerning references to graph methods, I'm not sure how to fix those but I'm hoping this might?

I'm hoping that merging these changes will reduce the number of errors quite a bit so that it will be a little easier to diagnose the remaining problems.

**Comment by hagberg** [comment]
I think there was a reason why we didn't want to use autoclass.  But I don't remember exactly why right now.  Maybe the autoclass format didn't present the info in the way we wanted so we used a more explicit approach of specifying each method.

**Comment by jfinkels** [comment]
Adding the automodule directives actually doesn't seem to fix the problem. Hmm I'll keep looking at this, don't merge.

**Comment by hagberg** [comment]
How do I set up local doc build for this that matches readthedocs tools?  Then I could test this locally.
It's some version of sphinx and tools right?

**Comment by SanketDG** [comment]
@hagberg You could install their theme and all (not sure about this), but the best possible experience would be to create a readthedocs site from your own fork https://github.com/networkx/networkx/wiki/readthedocs

**Comment by jfinkels** [comment]
This was fixed by #1795 I think.

**Pull request closed without merging**